filecoin-proofs/src/api/mod.rs

```diff
@@ -67,20 +67,77 @@ pub fn get_unsealed_range<T: Into<PathBuf> + AsRef<Path>, Tree: 'static + Merkle
     offset: UnpaddedByteIndex,
     num_bytes: UnpaddedBytesAmount,
 ) -> Result<UnpaddedBytesAmount> {
+    let f_in = File::open(&sealed_path)
+        .with_context(|| format!("could not open sealed_path={:?}", sealed_path.as_ref()))?;
+
+    let f_out = File::create(&output_path)
+        .with_context(|| format!("could not create output_path={:?}", output_path.as_ref()))?;
+
+    let buf_f_out = BufWriter::new(f_out);
+
+    unseal_range::<_, _, _, Tree>(
+        porep_config,
+        cache_path,
+        f_in,
+        buf_f_out,
+        prover_id,
+        sector_id,
+        comm_d,
+        ticket,
+        offset,
+        num_bytes,
+    )
+}
+
+/// Unseals the sector read from `sealed_sector` and returns the bytes for a
+/// piece whose first (unpadded) byte begins at `offset` and ends at `offset`
+/// plus `num_bytes`, inclusive. Note that the entire sector is unsealed each
+/// time this function is called.
+///
+/// # Arguments
+///
+/// * `porep_config` - porep configuration containing the sector size.
+/// * `cache_path` - path to the directory in which the sector data's Merkle Tree is written.
+/// * `sealed_sector` - a byte source from which we read sealed sector data.
+/// * `unsealed_output` - a byte sink to which we write unsealed, un-bit-padded sector bytes.
+/// * `prover_id` - the prover-id that sealed the sector.
+/// * `sector_id` - the sector-id of the sealed sector.
+/// * `comm_d` - the commitment to the sector's data.
+/// * `ticket` - the ticket that was used to generate the sector's replica-id.
+/// * `offset` - the byte index in the unsealed sector of the first byte that we want to read.
+/// * `num_bytes` - the number of bytes that we want to read.
+#[allow(clippy::too_many_arguments)]
+pub fn unseal_range<P, R, W, Tree>(
+    porep_config: PoRepConfig,
+    cache_path: P,
+    sealed_sector: R,
+    unsealed_output: W,
+    prover_id: ProverId,
+    sector_id: SectorId,
+    comm_d: Commitment,
+    ticket: Ticket,
+    offset: UnpaddedByteIndex,
+    num_bytes: UnpaddedBytesAmount,
+) -> Result<UnpaddedBytesAmount>
+where
+    P: Into<PathBuf> + AsRef<Path>,
+    R: Read,
+    W: Write,
+    Tree: 'static + MerkleTreeTrait,
+{
     ensure!(comm_d != [0; 32], "Invalid all zero commitment (comm_d)");
 
+    let mut sealed_sector = sealed_sector;
```

Comment:
Is there a difference between this and making the function parameters `mut`?

Reply:
Negative. I'll remove the shadowing let-bindings and make the parameters mutable.